### PR TITLE
Fixes #6454: networks.conf empty on relay

### PR DESCRIPTION
**Pull request by VinceMacBuche**
http://www.rudder-project.org/redmine/issues/6454

### CODE REVIEW

**Comment by VinceMacBuche** [comment]
Something need to be done for the "policy_server" class too, but it should be done in another issue I think

File: rudder-core/src/main/scala/com/normation/rudder/domain/policies/RuleTarget.scala [hunk 1]
```diff
@@ -232,12 +232,11 @@ object RuleTarget extends Loggable {
       case AllTargetExceptPolicyServers => nodes ++ allNodes.collect { case(k,n) if(!n._1) => k }
       case PolicyServerTarget(nodeId) => nodes + nodeId
       case AllServersWithRole =>
-        //we have a special case for the root node that always belongs to that group, even if some weird
-        //scenario lead to the removal (or non addition) of them
-        nodes ++ allNodes.collect { case(k,n) if(n._2.size>0 || n._1 == Constants.ROOT_POLICY_SERVER_ID) => k }
+        // all nodes with server roles or policyServer (ie relay)
+        nodes ++ allNodes.collect { case(k,n) if(n._2.size>0 || n._1) => k }
```
Comment (ncharles) [inline rudder-core/src/main/scala/com/normation/rudder/domain/policies/RuleTarget.scala]: If I understand correctly, you are always adding k in the node; for there always is a n._1
this looks plain wrong

Comment (VinceMacBuche) [inline rudder-core/src/main/scala/com/normation/rudder/domain/policies/RuleTarget.scala]: N._1 is a boolean meaning that the node is a policyServer ( either the root server or a relay)

Maybe I should explicit the tuple here by having ( nodeId, ( isPolicyServer, server roles)) inster of (k,n)

The code before was completely wrong ( comparing a boolean and a string)

Comment (ncharles) [inline rudder-core/src/main/scala/com/normation/rudder/domain/policies/RuleTarget.scala]: ho, so it was before in another file ( see PR https://github.com/Normation/rudder/pull/824/files ) and that broke it.

**Comment by VinceMacBuche** [comment]
PR updated

**Pull request merged**